### PR TITLE
Move buildable module test to a module test

### PR DESCRIPTION
**Pull request by xster**
Move test out of add-to-app test since the add-to-app project is already materialized.

### CODE REVIEW

File: dev/integration_tests/ios_add2app/build_and_test.sh [hunk 1]
```diff
@@ -5,13 +5,7 @@ set -e
...
 
 pushd flutterapp
-../../../../bin/flutter build ios --debug --no-codesign -v
-
-pushd .ios
-xcodebuild -workspace Runner.xcworkspace -scheme Runner -destination generic/platform=iOS -configuration Debug CODE_SIGNING_REQUIRED=NO CODE_SIGNING_IDENTITY="" CODE_SIGNING_ALLOWED=NO
-xcodebuild -workspace Runner.xcworkspace -scheme Runner -destination generic/platform=iOS -configuration Profile CODE_SIGNING_REQUIRED=NO CODE_SIGNING_IDENTITY="" CODE_SIGNING_ALLOWED=NO
-xcodebuild -workspace Runner.xcworkspace -scheme Runner -destination generic/platform=iOS -configuration Release CODE_SIGNING_REQUIRED=NO CODE_SIGNING_IDENTITY="" CODE_SIGNING_ALLOWED=NO
-popd
+../../../../bin/flutter build ios --debug --simulator --no-codesign
```
Comment (xster) [inline dev/integration_tests/ios_add2app/build_and_test.sh]: You're testing on a simulator below anyway. Build x86 to make things marginally faster.

**Review by jmagman — COMMENTED** [review]
Does module_test_ios run anywhere?  I don't even see a integration_tests-macos shard.

**Comment by jmagman** [comment]
> Does module_test_ios run anywhere? I don't even see a integration_tests-macos shard.

Guess not.
https://github.com/flutter/flutter/blob/master/.cirrus.yml#L501

Can we diagnose the flakiness?  Or make this the only test that runs in a new integration_tests-macos shard?  Or was module_test_ios itself the cause of the flakiness?
I'm afraid this will atrophy again if it's not running on any CI system.

**Comment by xster** [comment]
T.T

https://github.com/flutter/flutter/issues/36120

**Review by dnfield — APPROVED** [review]
LGTM

**Review by jmagman — APPROVED** [review]

**Comment by xster** [comment]
Gonna rerun this on head to make sure this doesn't flake on cirrus

**Comment by xster** [comment]
These mac tests take forever to schedule. But didn't flake on reruns. Merging.

**Pull request merged**